Project 5 For John Beverly

```diff
@@ -59,345 +59,388 @@ Group 2 contains Emanuel, Katie, Josh V (commonly know as V by group member), Ca
+sh:datatype xsd:boolean ;    
 
-sh:path schema:datePublished ;        
+sh:in ( true )  ] ;
 
-sh:datatype xsd:date ;        
+sh:property [    
 
-sh:message "The album release date must be a valid date in the format YYYY-MM-DD."    ] ;
+sh:path ex:hasType ;    
 
+sh:in ( "asymmetric" )  
 
-sh:property [        
+] .
 
-sh:path schema:numTracks ;        
+9.	Constraint for disjoint of symmetry and asymmetry 
+@prefix sh: <http://www.w3.org/ns/shacl#> .
 
-sh:datatype xsd:integer ;        
+@prefix ex: <http://example.com/> .
 
-sh:minInclusive 1 ;        
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
-sh:message "The album must have at least 1 track."   
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
-] ;
+ex:symmetricalPropertyShape
 
-sh:property [       
+a sh:PropertyShape ;
 
-sh:path schema:recordLabel ;       
+sh:path ex:symmetricalProperty ;
 
-sh:datatype xsd:string ;       
+sh:propertyConstraint [
 
-sh:message "The album record label must be a string."   
+sh:symmetrical true ;
 
 ] .
 
 
-# Define the shape for the artist object
+ex:asymmetricalPropertyShape
+
+a sh:PropertyShape ;
 
-ex:ArtistShape
+sh:path ex:asymmetricalProperty ;
+
+sh:propertyConstraint [
+
+sh:asymmetrical true ;
+
+] .
+
+ex:MyClassShape
 
 a sh:NodeShape ;
 
-sh:targetClass schema:MusicGroup ;
 
-sh:property [        
+sh:property [
 
-sh:path schema:name ;        
+sh:xoneOf ( ex:symmetricalPropertyShape ex:asymmetricalPropertyShape ) ;
 
-sh:datatype xsd:string ;       
+] .
 
-sh:minLength 3 ;        
 
-sh:message "The artist name must be a string of at least 3 characters."   
 
-] ;
-    sh:property [        
+10.	Constraint for PropertyIn for Asymmetrical and Irrelexive 
+@prefix sh: <http://www.w3.org/ns/shacl#> .
 
-sh:path schema:genre ;        
+@prefix ex: <http://example.com/> .
 
-sh:datatype xsd:string ;       
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
-sh:message "The artist genre must be a string."   
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
-] .
 
+ex:asymmetricalPropertyShape
 
-# Define the shape for the user object
+a sh:PropertyShape ;
 
-ex:UserShape
+sh:path ex:asymmetricalProperty ;
 
-a sh:NodeShape ;
+sh:propertyConstraint [
 
-sh:targetClass schema:Person ;
+sh:asymmetrical true ;
 
-sh:property [        
+] .
 
-sh:path schema:name ;        
+ex:irreflexivePropertyShape
 
-sh:datatype xsd:string ;       
+a sh:PropertyShape ;
 
-sh:minLength 3 ;        
+sh:path ex:irreflexiveProperty ;
 
-sh:message "The user name must be a string of at least 3 characters."   
+sh:propertyConstraint [
 
-] ;
+sh:irreflexive true ;
 
-sh:property [        
+] .
 
-sh:path schema:email ;        
+ex:MyClassShape
 
-sh:datatype xsd:string ;        
+a sh:NodeShape ;
 
-sh:pattern "[^@]+@[^.]+\\..+" ;        
+sh:property [
 
-sh:message "The user email must be a valid email address."   
+sh:or ( ex:asymmetricalPropertyShape ex:irreflexivePropertyShape ) ;
 
 ] .
 
 
+
```

Comment:
Hi all, sorry for sending this review late but I missed that you updated the files with this new stuff. I have a general concern, that is, you seem to be using in multiple occasions constrains of the type

sh:property [    

sh:path rdf:value ;    

sh:minCount 1 ;    

sh:maxCount 1 ;    

Not sure how this would work in enforcing any logical property for the relation(s) in question, or if it should be repeated in different shapes. I have added some other comments in the text. If I am missing something about the purpose of the shapes or how they work, then perhaps I might suggest to add some explanation about it (of couse, it could also be that I am missing the general sense because I am not sleeping these days). Hope this can be of help!

Giacomo

Reply:
Giacomo,
Please please please make getting enough rest a priority for yourself :) you deserve to rest! I appreciated all of your comments and I agree the min/maxCount is a reoccurring issue I will set out to resolve! However, I do want to note that this structure will not disappear as it is important for a constraint (at least ours) to be able to constraint for the number of instances to not be infinite. We are grateful for your time and efforts on our project!
Get some sleep, please!
Peace and wellness,
Group2